### ISSUE TITLE
Multiple queue architectures within Admiral

### ISSUE THREAD

**Issue by jdrydn** [car-throttle/admiral#3]
I think it would be nice if `admiral` could also handle simple FIFO queues as well has these repeating queues, in a similar style to other Node libraries, since the majority of it would be (testable) logic with few-to-none dependencies required (unlike others that literally pull in templating dependencies for pieces of tools we don't even use!)

Imagine:
```
const admiral = require('admiral'); // src/admiral.js

const cycling = admiral.createCycleQueue(opts); // src/cycle-queue.js
const fifo = admiral.createFifoQueue(opts); // src/fifo-queue.js

cycling.create('count-something', 'id-1').then(...).catch(...);

fifo.push('new-email', { to: 'human@someimportantcompany.com', ... }).then(...).catch(...);

cycling.process('count-something', job => countSomethingJob(job));
fifo.process('new-email', job => emailSomeone(job));
```

I've been thinking about this for a while, on one hand there's no point in reinventing the wheel and on the other hand it would be nice if Admiral was capable of different queue architectures!

**Comment by jdrydn** [comment]
Thoughts @g-wilson?

**Comment by g-wilson** [comment]
I agree, I think it would be useful to have more novel job orchestration patterns.

FIFO might be useful if the developer wishes to combine a simple FIFO process in addition to the normal Admiral "repeating tasks" functionality.
